presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache;
+
+import com.facebook.presto.cache.alluxio.AlluxioCachingFileSystem;
+import com.facebook.presto.cache.basic.BasicCachingFileSystem;
+import com.facebook.presto.hive.filesystem.ExtendedFileSystem;
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+
+import static com.facebook.presto.cache.CacheType.ALLUXIO;
+
+public class CacheFactory
+{
+    CachingFileSystem createCachingFileSystem(
+            Configuration factoryConfig,
+            URI factoryUri,
+            ExtendedFileSystem fileSystem,
+            CacheManager cacheManager,
+            CacheConfig cacheConfig)
+    {
+        if (cacheConfig.isCachingEnabled() && cacheConfig.getCacheType() == ALLUXIO) {
+            return new CachingFileSystem(new AlluxioCachingFileSystem(factoryConfig,
```

Comment:
- CachingFileSystem and AlluxioCachingFileSystem are in parallel rather than AlluxioCachingFileSystem is the cache tier for CachingFileSystem
- CachingFileSystem should be renamed to LocalRangeCachingFileSystem to be more clear

Reply:
I think the idea here was to keep an abstraction even thought its dummy at the moment for potential future use-cases. 
The hierarchy is like this:
CachingFIleSystem 
	AlluxioCachingFileSystem
	BasicCachingFileSystem (should be renamed, which is old CachingFileSystem)

Reply:
That is fine too. So let's rename `BasicCachingFileSystem` to `LocalRangeCachingFileSystem` to keep it sync with the CacheType naming

Reply:
Lets rename BasicCachingFileSystem to GenericCachingFIleSystem.

Reply:
I still have a hard time understanding how the wrapping here could benefit potential future use case rather than making caching file system an interface. Are we saying `CachingFIleSystem` is going to provide some generic cache feature, and the delegate caching FS would have more sophisticated logic? If so, maybe we should put cache flag check in `CachingFIleSystem`?

Reply:
I think it does provide few benefits. The first is that it gives an annotation to all filesystems(we will have more inn future) used for caching. The 2nd is it provides a place to perform common logic here. One example could be to check if cache is enabled or not itself. Is that what you mean by put cache flag check here?

Reply:
@shixuan-fan `CachingFileSystem` is currently just a simple wrapper class serving two purposes:
(1) delegating everything to caching FS
(2) serve as base class for potential extension or type-safety check.

Which part are you questioning?

Reply:
@jainxrohit @apc999 I just feel like currently this base class is a bit redundant for the sake of current commit scope, and not sure if we should actually introduce it at the time it is really needed. No strong opinion.

Agreed that there could be some common logic for caching related logic, say, maybe per table cache quota tracking? I'm not sure, any idea in mind?